daemon/cmd/agenthealth.go

```diff
@@ -42,6 +42,7 @@ func (d *Daemon) startAgentHealthHTTPService() {
 		statusCode := http.StatusOK
 		sr := d.getStatus(true)
 		if isUnhealthy(&sr) {
+			log.WithField("state", sr.Cilium.State).Warnf("/healthz returning unhealthy: %s", sr.Cilium.Msg)
```

Comment:
Minor nit: we have `pkg/logging/logfields` for well-known / hardcoded log fields; could you check if it's already defined there then add it if not?

Reply:
@christarazi added a `State` logfield in that package as suggested.